e2e/webhook_test.go

```diff
@@ -0,0 +1,122 @@
+	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/kube"
+	"github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+// TestWebhooksNoRBAC validates that the operator works without any webhook RBAC policies.
+func TestWebhooksNoRBAC(t *testing.T) {
+	ctx := context.Background()
+	clientSet, _, err := setupCluster(ctx, t)
+	if err != nil {
+		t.Fatalf("error instantiating clients. err: %s", err)
+	}
+
+	if err := clientSet.RbacV1().ClusterRoles().Delete(ctx, "gmp-system:operator:webhook-admin", metav1.DeleteOptions{}); err != nil {
+		t.Fatalf("error deleting cluster role: %s", err)
+	}
+	if err := clientSet.RbacV1().ClusterRoleBindings().Delete(ctx, "gmp-system:operator:webhook-admin", metav1.DeleteOptions{}); err != nil {
+		t.Fatalf("error deleting cluster role binding: %s", err)
+	}
+
+	// Restart the GMP operator could already be healthy before we deleted the RBAC policies.
+	t.Log("restarting operator")
+	if err := deploymentRestart(ctx, clientSet, operator.NameOperator, operator.DefaultOperatorNamespace); err != nil {
+		t.Fatalf("error restarting operator. err: %s", err)
+	}
+
+	restConfig, err := newRestConfig()
+	if err != nil {
+		t.Fatalf("error creating rest config: %s", err)
+	}
+
+	kubeClient, err := newKubeClient(restConfig)
+	if err != nil {
+		t.Fatalf("error creating client: %s", err)
+	}
+
+	t.Log("waiting for operator to be deployed")
+	if err := kube.WaitForDeploymentReady(ctx, kubeClient, operator.DefaultOperatorNamespace, operator.NameOperator); err != nil {
+		t.Fatalf("error waiting for operator deployment to be ready: %s", err)
+	}
+
+	t.Log("waiting for operator to be ready")
+	if err := wait.PollUntilContextCancel(ctx, 3*time.Second, true, func(ctx context.Context) (bool, error) {
+		logs, err := deploy.OperatorLogs(ctx, kubeClient, clientSet, operator.DefaultOperatorNamespace)
+		if err != nil {
+			t.Logf("unable to get operator logs: %s", err)
+			return false, nil
+		}
+		t.Logf("waiting for operator logs to contain start message")
+		return strings.Contains(logs, "starting GMP operator"), nil
+	}); err != nil {
+		t.Fatalf("unable to check operator ready: %s", err)
+	}
+
+	if err := wait.PollUntilContextCancel(ctx, 3*time.Second, true, func(ctx context.Context) (bool, error) {
+		logs, err := deploy.OperatorLogs(ctx, kubeClient, clientSet, operator.DefaultOperatorNamespace)
+		if err != nil {
+			t.Logf("unable to get operator logs: %s", err)
+			return false, nil
+		}
+
+		t.Logf("waiting for operator logs to contain RBAC message")
+		return strings.Contains(logs, "delete legacy ValidatingWebHookConfiguration was not allowed"), nil
+	}); err != nil {
+		t.Fatalf("unable to check operator logs: %s", err)
+	}
+}
+
+func deploymentRestart(ctx context.Context, kubeClient kubernetes.Interface, name, namespace string) error {
```

Comment:
IIUC when the pod is deleted, it gets recreated automatically? can you please add a comment here.